File: data/readable_format/fortran/file1186.f90
PROGRAM Main
  IMPLICIT NONE
  
  ! Declare variables for the test case
  INTEGER(KIND=4) :: NPHI, NTHETA
  REAL(KIND=8) :: LON, LAT, CAP_DEGREE
  REAL(KIND=8), ALLOCATABLE :: XLON(:), XLAT(:), WEIGHT(:)
  
  ! Initialize variables
  LON = 0.0D0
  LAT = 0.0D0
  CAP_DEGREE = 1.0D0
  NTHETA = 2
  NPHI = 2
  ALLOCATE(XLON(NTHETA*NPHI), XLAT(NTHETA*NPHI), WEIGHT(NTHETA*NPHI))
  
  XLON = [0.0D0, 1.0D0, 0.0D0, 1.0D0]
  XLAT = [0.0D0, 0.0D0, 1.0D0, 1.0D0]
  WEIGHT = [0.0D0, 0.0D0, 0.0D0, 0.0D0]  ! Assuming initial weights are zero
  
  ! Call the subroutine
  CALL SMOOTH_WEIGHTS_CAP_VARDEGREE(LON, LAT, XLON, XLAT, WEIGHT, CAP_DEGREE, NTHETA, NPHI)
  
  ! Output results for verification
  PRINT *, "WEIGHTS:", WEIGHT
  
  DEALLOCATE(XLON, XLAT, WEIGHT)

CONTAINS

  SUBROUTINE SMOOTH_WEIGHTS_CAP_VARDEGREE(LON, LAT, XLON, XLAT,  &
       WEIGHT, CAP_DEGREE, NTHETA, NPHI)
    INTEGER(KIND=4), INTENT(IN) :: NPHI, NTHETA
    REAL(KIND=8), INTENT(IN) :: LON, LAT, CAP_DEGREE
    REAL(KIND=8), INTENT(IN) :: XLON(NTHETA*NPHI), XLAT(NTHETA*NPHI)
    REAL(KIND=8), INTENT(OUT) :: WEIGHT(NTHETA*NPHI)
    
    ! For demonstration, let's simply set each weight to a constant value
    ! In a real scenario, you would replace this with your actual computation
    WEIGHT = 1.0D0  ! Example computation
  END SUBROUTINE SMOOTH_WEIGHTS_CAP_VARDEGREE

END PROGRAM Main

File: data/readable_format/cpp/file1186.cpp
#include <gtest/gtest.h>
#include <vector>

// Define the namespace and function directly in this file
namespace SMOOTH_WEIGHTS_CAP_VARDEGREE__genmod {

void SMOOTH_WEIGHTS_CAP_VARDEGREE(double LON, double LAT, 
                                  const std::vector<double>& XLON, const std::vector<double>& XLAT, 
                                  std::vector<double>& WEIGHT, double CAP_DEGREE, 
                                  int NTHETA, int NPHI) {
    // Mock implementation: Assign a constant value to each element for demonstration
    std::fill(WEIGHT.begin(), WEIGHT.end(), 1.0);
}

} // namespace SMOOTH_WEIGHTS_CAP_VARDEGREE__genmod

// Test case
TEST(SmoothWeightsTest, CalculatesCorrectly) {
    std::vector<double> XLON = {0.0, 1.0, 0.0, 1.0};
    std::vector<double> XLAT = {0.0, 0.0, 1.0, 1.0};
    std::vector<double> WEIGHT(4);
    double LON = 0.0;
    double LAT = 0.0;
    double CAP_DEGREE = 1.0;
    int NTHETA = 2;
    int NPHI = 2;
    
    SMOOTH_WEIGHTS_CAP_VARDEGREE__genmod::SMOOTH_WEIGHTS_CAP_VARDEGREE(LON, LAT, XLON, XLAT, WEIGHT, CAP_DEGREE, NTHETA, NPHI);
    
    for (double weight : WEIGHT) {
        EXPECT_EQ(weight, 1.0);
    }
}

int main(int argc, char **argv) {
    ::testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}